src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java

```diff
@@ -344,6 +349,51 @@ protected AstNode literal() throws ScanException, ParseException {
     return super.literal();
   }
 
+  @Override
+  protected AstNode cmp(boolean required) throws ScanException, ParseException {
+    AstNode v = add(required);
+    if (v == null) {
+      return null;
+    }
+    while (true) {
+      switch (getToken().getSymbol()) {
+        case LT:
+          consumeToken();
+          v = createAstBinary(v, add(true), AstBinary.LT);
+          break;
+        case LE:
+          consumeToken();
+          v = createAstBinary(v, add(true), AstBinary.LE);
+          break;
+        case GE:
+          consumeToken();
+          v = createAstBinary(v, add(true), AstBinary.GE);
+          break;
+        case GT:
+          consumeToken();
+          v = createAstBinary(v, add(true), AstBinary.GT);
+          break;
+        case EXTENSION:
+          if (getExtensionHandler(getToken()).getExtensionPoint() == ExtensionPoint.CMP) {
+            v = getExtensionHandler(consumeToken()).createAstNode(v, add(true));
+            break;
+          }
+        default:
```

Comment:
This has a smell to me - I haven't spent any time in this parser code, but I don't like duplicating this much code to make the override work. Is it possible to create a two-word token ("not in")?

Another kernel of thought in how we approach this might be to rewrite `foo not in bar` to `not (foo in bar)`, but that has a different smell. Just sharing it here in case that gets you thinking, though

Reply:
It does have a bit of a smell, but the original Parser code doesn't provide more options for extending here. Because of the call to `add()` there isn't a way to use `super` calls and add on to the original code. Seems we've had to do this duplication in the past, for example with the `value()` method. I don't think there's a way that we can have a two-word token unfortunately because of the way that tokens are scanned.

While I like the idea of simply `not (foo in bar)`, to get parity here with Jinja, I want to add functionality in Jinjava for:
> The is and in operators support negation using an infix notation, too: `foo is not bar` and `foo not in bar`